include/ocvsmd/sdk/defines.hpp

```diff
@@ -115,6 +122,18 @@ enum class CyphalPriority : std::uint8_t
 
 };  // CyphalPriority
 
+/// Defines helper which owns mutable raw data buffer.
+///
+struct OwnMutablePayload
+{
+    /// Holds the size of the raw data buffer. It could be less than it was allocated.
+    std::size_t size;
+
+    /// Holds smart pointer to the raw data buffer.
+    std::unique_ptr<cetl::byte[]> data;  // NOLINT(*-avoid-c-arrays)
```

Comment:
@thirtytwobits docs matter, but redundant docs are worse than none at all. Can we avoid documenting entities whose purpose is obvious from definition?

Reply:
I haven't been a pedant on this. I always ask for class docs and public methods but have been flexible otherwise. These five lines are a good example in that line 129's documentation is actually quite useful as it adds information that isn't available in the datatype. Line 132 is redundant unless we think the reader would confuse this byte buffer for something other than "the raw data buffer" (context matters).